### ISSUE TITLE
Auth Attempt Failed: Public Key Does Not Exist

### ISSUE THREAD

**Issue by 509dave16** [amark/gun#574]
## TLDR
@amark  I think the issue stems from the gunRoot not being populated with 'public key': 'user' pairs. query.js looks for those. After create.js puts one on the root, I can see it on my server peer and my browser peer. But for some reason on refresh, the direct children of the root for public keys are not there anymore on the browser peer.

## Expected behavior
When executing `gun.user().auth(<username>,<password>, cb)`, I expect that a previously created user should be logged in.

## Resulting behavior
<img width="1792" alt="auth_attempt_failed_public_key_does_not_exist" src="https://user-images.githubusercontent.com/2925882/42554904-669c1844-849b-11e8-9912-da5b5d47241e.png">

## Troubleshooting Information
- **Source File Where Error Occurs**: File where error is originally thrown: https://github.com/amark/gun/blob/master/sea/query.js
- **Line That Fails To Do Its Job**: https://github.com/amark/gun/blob/master/sea/query.js#L30
- **Because Of This Line(offender)**: https://github.com/amark/gun/blob/master/sea/query.js#L26

## Hypothesis
- `gunRoot.get(pub).get((at, ev) => {`(offender) is trying to access a path where there is no put value.
-  Somehow between refreshes after a creating a user the nodes that should be there for public keys are gone

**Comment by 509dave16** [comment]
@amark Found a work around. The way that recall is handled is interesting but there's problems with `gun.user().recall({ sessionStorage: true});` in how the promise settles and the fact that a callback is not passed to the `gun.user().auth(...)`.  So I pass the `recall` option and re-implement what recall does for auth.
```javascript
// Instantiating Gun
const gun = Gun({
	peers: ['http://localhost:8082/gun'],
	recall: true
});
...
// Elsewhere in whatever is used to check for authentication
public async isAuthenticated(): Promise<any> {
	const tmp = window.sessionStorage;
		if (tmp) {
			if (tmp.alias && tmp.tmp) {
			try {
			        await this.login({ username: tmp.alias, password: tmp.tmp});
				return true;
			} catch (e) {
				console.log(e);
			}
		}
	}
	return false;
}
```
I'm hoping to contribute to helping address whatever issues currently exist with SEA for auth. Let me know if what above doesn't make sense.

**Comment by 509dave16** [comment]
@amark Closing this for now. It go muddied with different things. Here's the gist of what I have done to make this work by doing my own storage management.
https://gist.github.com/509dave16/ee952c687424822dbc667cae1d71cff7

**Comment by amark** [comment]
@509dave16 excellently reported, thank you for the thoroughness, it is super appreciated/helpful.

And also super thanks for dealing with workarounds for now.

`sessionStorage` should persist upon refresh (but not exit, and come back later).

It seems to me that maybe it isn't `.recall` that is failing, but my comment in https://github.com/amark/gun/issues/579 ?

**Comment by jadbox** [comment]
@509dave16 did you find out how you solved the "Public key is missing" error? I can't seem to get past it.

**Comment by 509dave16** [comment]
@jadbox I avoided using `recall`. So stored the credentials in the session myself. And then would re-authenticate on every page if the user wasn't already logged in.
